randovania/server/multiplayer/world_api.py

```diff
@@ -174,42 +209,30 @@ def world_sync(sio: ServerApp, request: ServerSyncRequest) -> ServerSyncResponse
     sessions_to_update_meta = set()
 
     for uid, world_request in request.worlds.items():
-        try:
+        with sentry_sdk.start_transaction(op="message", name="sync_one_world") as span:
+            span.set_tag("world_id", uid)
+            span.set_data("message.error", 0)
+
             try:
-                world = World.get_by_uuid(uid)
-            except peewee.DoesNotExist:
-                raise error.WorldDoesNotExistError()
-
-            _check_user_is_associated(user, world)
-
-            if world_request.status == GameConnectionStatus.Disconnected:
-                flask_socketio.leave_room(_get_world_room(world))
-            else:
-                flask_socketio.join_room(_get_world_room(world))
-                worlds_to_update.add(world)
-                sio.store_world_in_session(world)
-
-            session_id = update_association(user, world, world_request.inventory, world_request.status)
-            if session_id is not None:
-                sessions_to_update_meta.add(session_id)
-
-            if world_request.request_details:
-                world_details[uid] = ServerWorldResponse(
-                    world_name=world.name,
-                    session=world.session.create_list_entry(),
-                )
-
-            # Do this last, as it fails if session is in setup
-            if world_request.collected_locations:
-                worlds_to_update.update(collect_locations(sio, world, world_request.collected_locations))
-
-        except error.BaseNetworkError as e:
-            logger().info("Failed sync for %s: %s", uid, e)
-            failed_syncs[uid] = e
-
-        except Exception as e:
-            logger().exception("Failed sync for %s: %s", uid, e)
-            failed_syncs[uid] = error.ServerError()
+                response, session_id, new_worlds_to_update = sync_one_world(sio, user, uid, world_request)
+
+                if response is not None:
+                    world_details[uid] = response
+
+                if session_id is not None:
+                    sessions_to_update_meta.add(session_id)
+
+                worlds_to_update.update(new_worlds_to_update)
```

Comment:
Why we need to do this again? This is already done in `sync_one_world` in `if world_request.collected_locations`.

Reply:
Two different `worlds_to_update` variables!